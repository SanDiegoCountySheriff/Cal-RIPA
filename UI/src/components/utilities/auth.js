import { PublicClientApplication } from '@azure/msal-browser'

export const doSignIn = async () => {
  await this.$msalInstance
    .loginPopup({})
    .then(() => {
      const myAccounts = this.$msalInstance.getAllAccounts()
      this.account = myAccounts[0]
      this.$emitter.emit('login', this.account)
    })
    .catch(error => {
      console.error(`error during authentication: ${error}`)
    })
}
