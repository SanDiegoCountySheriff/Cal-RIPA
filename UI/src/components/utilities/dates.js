export const formatDate = dateStr => {
  if (dateStr && dateStr.length > 0) {
    const str = dateStr.replace(/-/g, '')
    const year = str.substring(0, 4)
    const month = str.substring(4, 6)
    const day = str.substring(6, 8)
    return `${year}-${month}-${day}`
  }

  return ''
}
