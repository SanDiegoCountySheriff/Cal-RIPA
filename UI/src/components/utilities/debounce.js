export const debounce = (func, delay, immediate) => {
  let timerId
  return (...args) => {
    const boundFunc = func.bind(this, ...args)
    clearTimeout(timerId)
    if (immediate && !timerId) {
      boundFunc()
    }
    const calleeFunc = immediate
      ? () => {
          timerId = null
        }
      : boundFunc
    timerId = setTimeout(calleeFunc, delay)
  }
}
