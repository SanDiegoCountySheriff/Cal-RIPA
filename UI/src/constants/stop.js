export const SUBMISSION_STATUSES = [
  {
    text: 'Failed',
    value: 'Failed',
    editable: true,
  },
  {
    text: 'Submitted',
    value: 'Submitted',
    editable: false,
  },
  {
    text: 'Unsubmitted',
    value: 'Unsubmitted',
    editable: true,
  },
  {
    text: 'Resubmitted',
    value: 'Resubmitted',
    editable: false,
  },
]
