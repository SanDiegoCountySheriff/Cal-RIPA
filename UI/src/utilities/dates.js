import differenceInHours from 'date-fns/differenceInHours'
import { differenceInYears as diffInYears } from 'date-fns'
import format from 'date-fns/format'
import isValid from 'date-fns/isValid'
import endOfDay from 'date-fns/endOfDay'

export const formatDate = dateStr => {
  if (dateStr && dateStr.length > 0) {
    const str = dateStr.replace(/-/g, '')
    const year = str.substring(0, 4)
    const month = str.substring(4, 6)
    const day = str.substring(6, 8)
    return `${year}-${month}-${day}`
  }

  return ''
}

export const isValidDate = dateStr => {
  const date = parseDate(dateStr, '00:00')
  return isValid(date)
}

const parseDate = (dateStr, timeStr) => {
  return new Date(`${dateStr}T${timeStr}`)
}

export const dateNotInFuture = (dateStr, timeStr) => {
  const date = parseDate(dateStr, timeStr || '00:00')
  return new Date().getTime() >= date.getTime()
}

export const differenceInYears = date => {
  const diff = diffInYears(new Date(), new Date(date))
  return diff
}

export const dateWithinLastHours = (dateStr, timeStr, hours) => {
  const date = parseDate(dateStr, timeStr)
  const diff = differenceInHours(new Date(), date)
  return diff < hours
}

export const formatDateTime = (dateStr, timeStr) => {
  return new Date(`${dateStr}T${timeStr}`)
}

export const formatShortDate = dateStr => {
  const date = parseDate(dateStr, '00:00')
  return format(new Date(date), 'yyyy-MM-dd')
}

export const uniqueId = () => {
  return format(new Date(), 'yyyyMMddkkmm')
}

export const formatToIsoCurrentDate = () => {
  return format(endOfDay(new Date()), `yyyy-MM-dd'T'HH:mm:ss.SSSxxx`)
}

export const formatToIsoDate = date => {
  return format(endOfDay(date), `yyyy-MM-dd'T'HH:mm:ss.SSSxxx`)
}
