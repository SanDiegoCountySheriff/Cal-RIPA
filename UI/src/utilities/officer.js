export const getOfficerStartDate = () => {
  const startDate = localStorage.getItem('ripa_officer_start_date')
  return startDate || null
}

export const getOfficerYearsExperience = () => {
  const yearsExperience = localStorage.getItem('ripa_officer_years_experience')
  return +yearsExperience || null
}

export const getOfficerAssignment = () => {
  const assignment = localStorage.getItem('ripa_officer_assignment')
  return +assignment || null
}

export const getOfficerOtherType = () => {
  const otherType = localStorage.getItem('ripa_officer_other_type')
  return otherType || null
}
