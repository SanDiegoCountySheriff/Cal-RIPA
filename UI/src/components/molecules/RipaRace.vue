<template>
  <div class="ripa-race tw-pb-4">
    <ripa-form-header
      title="Perceived Race or Ethnicity"
      required
      subtitle="ยง999.226(a)(4)"
      v-on="$listeners"
    >
    </ripa-form-header>

    <v-container>
      <v-row no-gutters>
        <v-col cols="12" sm="12">
          <ripa-check-group
            v-model="model.person.perceivedRace"
            :disabled="disabled"
            :items="raceItems"
            :rules="raceRules"
          >
          </ripa-check-group>
        </v-col>
      </v-row>
    </v-container>
  </div>
</template>

<script>
import RipaFormHeader from '@/components/molecules/RipaFormHeader'
import RipaCheckGroup from '@/components/atoms/RipaCheckGroup'
import { RACES } from '@/constants/form'

export default {
  name: 'ripa-race',

  components: {
    RipaCheckGroup,
    RipaFormHeader,
  },

  data() {
    return {
      raceItems: RACES,
    }
  },

  computed: {
    model: {
      get() {
        return this.value
      },
      set(newVal) {
        this.$emit('input', newVal)
      },
    },

    raceRules() {
      const options = this.model.person.perceivedRace
      return [options.length > 0 || 'At least one race is required']
    },
  },

  watch: {
    model: {
      handler: function (newVal) {
        this.model = newVal
      },
      deep: true,
    },
  },

  props: {
    value: {
      type: Object,
      default: () => {},
    },
    disabled: {
      type: Boolean,
      default: false,
    },
  },
}
</script>
