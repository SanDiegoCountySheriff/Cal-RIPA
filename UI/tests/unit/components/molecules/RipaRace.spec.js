import RipaRace from '@/components/molecules/RipaRace.vue'
import { shallowMount, mount } from '@vue/test-utils'
import { defaultStop } from '@/utilities/stop'
import Vuetify from 'vuetify'

describe('Ripa Race', () => {
  let vuetify
  let wrapper
  let stop

  beforeEach(() => {
    vuetify = new Vuetify()
    stop = defaultStop()
  })

  const factory = propsData => {
    return shallowMount(RipaRace, {
      vuetify,
      propsData: {
        ...propsData,
      },
    })
  }

  it('should match snapshot', () => {
    wrapper = mount(RipaRace, {
      vuetify,
      propsData: {
        value: stop,
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
