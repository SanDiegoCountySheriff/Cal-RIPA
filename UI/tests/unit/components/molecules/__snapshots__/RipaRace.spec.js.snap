// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ripa Race should match snapshot 1`] = `
<div class="ripa-race tw-pb-4">
  <div class="ripa-form-header">
    <div class="row tw-ml-1 tw-mt-1"><span class="v-tooltip v-tooltip--top"><!----><i aria-hidden="true" aria-haspopup="true" aria-expanded="false" class="v-icon notranslate mdi mdi-alert theme--light error--text" style="font-size: 16px;"></i></span> <button type="button" class="v-icon notranslate icon v-icon--link mdi mdi-scale-balance theme--light primary--text" style="font-size: 16px;"></button>
      <div class="ripa-header"><span class="tw-text-base md:tw-text-lg tw-uppercase">Perceived Race or Ethnicity</span></div>
    </div>
  </div>
  <div class="container">
    <div class="row no-gutters">
      <div class="col-sm-12 col-12">
        <div class="ripa-check-group">
          <div class="v-input v-input--hide-details theme--light v-input--selection-controls v-input--checkbox">
            <div class="v-input__control">
              <div class="v-input__slot">
                <div class="v-input--selection-controls__input"><i aria-hidden="true" class="v-icon notranslate mdi mdi-checkbox-blank-outline theme--light"></i><input aria-checked="false" id="input-8" role="checkbox" type="checkbox" value="1">
                  <div class="v-input--selection-controls__ripple"></div>
                </div><label for="input-8" class="v-label theme--light" style="left: 0px; position: relative;">Asian</label>
              </div>
            </div>
          </div>
          <div class="v-input v-input--hide-details theme--light v-input--selection-controls v-input--checkbox">
            <div class="v-input__control">
              <div class="v-input__slot">
                <div class="v-input--selection-controls__input"><i aria-hidden="true" class="v-icon notranslate mdi mdi-checkbox-blank-outline theme--light"></i><input aria-checked="false" id="input-10" role="checkbox" type="checkbox" value="2">
                  <div class="v-input--selection-controls__ripple"></div>
                </div><label for="input-10" class="v-label theme--light" style="left: 0px; position: relative;">Black/African American</label>
              </div>
            </div>
          </div>
          <div class="v-input v-input--hide-details theme--light v-input--selection-controls v-input--checkbox">
            <div class="v-input__control">
              <div class="v-input__slot">
                <div class="v-input--selection-controls__input"><i aria-hidden="true" class="v-icon notranslate mdi mdi-checkbox-blank-outline theme--light"></i><input aria-checked="false" id="input-12" role="checkbox" type="checkbox" value="3">
                  <div class="v-input--selection-controls__ripple"></div>
                </div><label for="input-12" class="v-label theme--light" style="left: 0px; position: relative;">Hispanic/Latino(a)</label>
              </div>
            </div>
          </div>
          <div class="v-input v-input--hide-details theme--light v-input--selection-controls v-input--checkbox">
            <div class="v-input__control">
              <div class="v-input__slot">
                <div class="v-input--selection-controls__input"><i aria-hidden="true" class="v-icon notranslate mdi mdi-checkbox-blank-outline theme--light"></i><input aria-checked="false" id="input-14" role="checkbox" type="checkbox" value="4">
                  <div class="v-input--selection-controls__ripple"></div>
                </div><label for="input-14" class="v-label theme--light" style="left: 0px; position: relative;">Middle Eastern or South Asian</label>
              </div>
            </div>
          </div>
          <div class="v-input v-input--hide-details theme--light v-input--selection-controls v-input--checkbox">
            <div class="v-input__control">
              <div class="v-input__slot">
                <div class="v-input--selection-controls__input"><i aria-hidden="true" class="v-icon notranslate mdi mdi-checkbox-blank-outline theme--light"></i><input aria-checked="false" id="input-16" role="checkbox" type="checkbox" value="5">
                  <div class="v-input--selection-controls__ripple"></div>
                </div><label for="input-16" class="v-label theme--light" style="left: 0px; position: relative;">Native American</label>
              </div>
            </div>
          </div>
          <div class="v-input v-input--hide-details theme--light v-input--selection-controls v-input--checkbox">
            <div class="v-input__control">
              <div class="v-input__slot">
                <div class="v-input--selection-controls__input"><i aria-hidden="true" class="v-icon notranslate mdi mdi-checkbox-blank-outline theme--light"></i><input aria-checked="false" id="input-18" role="checkbox" type="checkbox" value="6">
                  <div class="v-input--selection-controls__ripple"></div>
                </div><label for="input-18" class="v-label theme--light" style="left: 0px; position: relative;">Pacific Islander</label>
              </div>
            </div>
          </div>
          <div class="v-input theme--light v-input--selection-controls v-input--checkbox">
            <div class="v-input__control">
              <div class="v-input__slot">
                <div class="v-input--selection-controls__input"><i aria-hidden="true" class="v-icon notranslate mdi mdi-checkbox-blank-outline theme--light"></i><input aria-checked="false" id="input-20" role="checkbox" type="checkbox" value="7">
                  <div class="v-input--selection-controls__ripple"></div>
                </div><label for="input-20" class="v-label theme--light" style="left: 0px; position: relative;">White</label>
              </div>
              <div class="v-messages theme--light">
                <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
